src/new-client/src/App/Footer/StatusButton/StatusButton.tsx

```diff
@@ -81,28 +85,81 @@ const selectAll = (event: SyntheticEvent) => {
   input.select()
 }
 
+const getInstanceNumber = (
+  record: Record<string, ServiceInstanceStatus>,
+  type: string,
+) => {
+  return Object.values(record).filter(
+    (x) => x.isConnected && x.serviceType === type,
+  ).length
+}
+
+const timeoutThreshold = 3000
+const requestTimeoutLogger$ = of(
+  ["blotter", "reference", "execution", "pricing", "analytics"].map(
+    (serviceType) =>
+      ({
+        serviceType,
+        connectionStatus: ServiceConnectionStatus.DISCONNECTED,
+        connectedInstanceCount: 0,
+      } as ServiceStatus),
+  ),
+)
+
+export const [useServiceStatus, serviceStatus$] = bind(
+  status$.pipe(
+    map((record) => {
+      const nameList: Set<string> = new Set()
+      const result: ServiceStatus[] = []
+      const recordValues = Object.values(record)
+      recordValues.forEach((next) => {
+        const newStatus = {
+          serviceType: next.serviceType,
+          connectedInstanceCount: getInstanceNumber(record, next.serviceType),
+          connectionStatus: getInstanceNumber(record, next.serviceType)
+            ? ServiceConnectionStatus.CONNECTED
+            : ServiceConnectionStatus.DISCONNECTED,
+        }
+        if (nameList.size !== nameList.add(newStatus.serviceType).size) {
+          result.push(newStatus)
+        }
+      })
+      return result
+    }),
+    timeoutWith(timeoutThreshold, requestTimeoutLogger$),
+  ),
+  [] as ServiceStatus[],
+)
+
+export const [useApplicationStatus, applicationStatus$] = bind(
+  serviceStatus$.pipe(map((status) => getApplicationStatus(status))),
```

Comment:
By the looks of it, it seems that we are conflating the connection status with the health of the services, but those are 2 different things:

- The connection status has to do with the socket connection: connected, disconnected, connecting...
- The health of the services has to do with whether the services are healthy or not, which I suppose it depends on whether there are more than zero instances of each up and running.

We should clarify what the business requirement for this button is: whether it has to display the connection status or the health of the services, or both.